### ISSUE TITLE
Consider replacing librsvg with libresvg

### ISSUE THREAD

**Issue by gunta** [libvips/libvips#1536]
https://github.com/RazrFalcon/resvg

Why:
- resvg substantially passes more tests than librsvg
- resvg is fast. Supports 4 rendering backends currently: cairo, skia, qt, raqote

This would help close many svg related issues.

The table of support is here: 
https://razrfalcon.github.io/resvg-test-suite/svg-support-table.html

Both library goals are different: librsvg is a GNOME library. librsvg is a cross-platform one with a multiple backend support.

**Comment by jcupitt** [comment]
Hello @gunta,

That's interesting, I hadn't heard of resvg. Sure, let's tag this as an enhancement and do some experiments.

**Comment by jcupitt** [comment]
@lovell I know you've been battling some SVG issues, you might be interested.

**Comment by lovell** [comment]
I'm rather ambivalent on this as librsvg is under heavy (re)development at the moment and is in a much better place than it was a couple of years ago when resvg was started.

It's worth noting that resvg:

* is removing support for "backends" other than raqote, which is not a bad thing, especially as librsvg is considering likewise - see https://github.com/RazrFalcon/resvg/issues/195 and https://gitlab.gnome.org/GNOME/librsvg/issues/530
* is not yet widely available via package manager - Debian unstable added it a year ago but hasn't updated recently

**Comment by gunta** [comment]
>> is removing support for "backends" other than raqote, which is not a bad thing, especially as librsvg is considering likewise

> According to my "benchmarks", raqote isn't that slow, but still considerably slower than Skia. I hope that this can be fixed soon. I do not plan to remove other backends right now.

So the change its not happening yet, but definitely Raqote, being used in Servo has good potential.

> is not yet widely available via package manager - Debian unstable added it a year ago but hasn't updated recently

Since neither library seems to aim for 100% full support, 
there will be always a [mismatch with the spec](https://razrfalcon.github.io/resvg-test-suite/svg-support-table.html), and users of libvips might need one or another depending on what the `svg` content looks like or performance-wise.

Since we will not have a clear winner anytime soon, I suggest to support both `librsvg` and `resvg` as a compile-time option.

In my case I need to use the `image` SVG element `External PNG` feature of the [spec](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/image) which `resvg` does support.

# resvg goals
> resvg can be used to render SVG files based on a static SVG Full 1.1 subset, excluding fonts support. 
In simple terms: no animations, scripting and fonts.
The core idea is to make a fast, portable, small, multiple backend library.

# librsvg non-goals:
> Implement every single SVG feature that is in the spec.
Implement scripting or external access to the SVG's DOM.
Implement support for CSS-based animations
Replace the industrial-strength SVG rendering machinery in modern web browsers.

**Comment by jcupitt** [comment]
Ah, I hadn't realized it was not yet widely packaged, thanks Lovell. I think without an easy way to test and deploy, any new libvips code is likely to bitrot.

Let's look at this again when resvg is packaged.
